src/core/components/hv-route/index.tsx

```diff
@@ -18,10 +160,59 @@ import React, { Text, View } from 'react-native';
  * - Retrieves the navigator element from the context
  * - Passes the props, context, and url to HvRouteInner
  */
-export default function HvRoute() {
+export default function HvRoute(props: Types.Props) {
+  const navProps: NavigationContext.NavigationContextProps | null = useContext(
+    NavigationContext.Context,
+  );
+  const navCache: NavigatorContext.NavigatorCache | null = useContext(
+    NavigatorContext.NavigatorMapContext,
+  );
+  if (!navProps || !navCache) {
+    throw new NavigatorService.HvRouteError('No context found');
+  }
+
+  // Retrieve the url from params or from the context
+  let url: string | undefined = props.route?.params?.url;
+  // Fragment urls are used to designate a route within a document
+  if (url && NavigatorService.isUrlFragment(url)) {
+    url = navCache.routeMap?.get(NavigatorService.cleanHrefFragment(url));
+  }
+
+  if (!url) {
+    // Use the route id if available to look up the url
+    if (props.route?.params?.id) {
+      url = navCache.routeMap?.get(props.route.params.id);
+    } else {
+      // Try to use the initial route for this <navigator>
+      const initialRoute: string | undefined = navCache.initialRouteName;
+      if (initialRoute) {
+        url = navCache.routeMap?.get(initialRoute);
+      }
+    }
+  }
+
+  // Fall back to the entrypoint url
+  url = url || navProps.entrypointUrl;
+
+  // Get the navigator element from the context
+  let element: TypesLegacy.Element | undefined;
+  if (props.route?.params.id) {
+    element = navCache.elementMap?.get(props.route?.params?.id);
+  }
+
   return (
-    <View>
-      <Text>ROUTE</Text>
-    </View>
+    <HvRouteInner
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...{ ...props, ...navProps, ...navCache }}
```

Comment:
ideally we should avoid prop spreading as it obfuscate what we pass from one component to another. But maybe here if we combine both component we might not need this

Reply:
We'll keep for now as it's a requirement of passing the props along with the context data into the class